proposals/1406_multiple-idps/README.md

```diff
@@ -0,0 +1,493 @@
+apiVersion: config.supervisor.pinniped.dev/v1alpha1
+metadata:
+  name: demo-federation-domain
+  namespace: supervisor
+spec:
+  issuer: https://issuer.example.com/demo-issuer
+  tls:
+    secretName: my-federation-domain-tls
+
+  # Everything below here is the new part.
+  identityProviders:
+
+  - displayName: ActiveDirectory for Admins
+    objectRef:
+      name: ad-for-admins
+      kind: ActiveDirectoryIdentityProvider
+      apiGroup: idp.supervisor.pinniped.dev/v1alpha1
+
+    # Transforms are optional and apply only to logins from this IDP in this FederationDomain.
+    transforms:
+
+       # Optionally define variables that will be available to the expressions below.
+       constants:
+          # Validations would check that these names are legal CEL variable names and are unique within this list.
+         - name: prefix
+           type: string
+           stringValue: "ad:"
+         - name: onlyIncludeGroupsWithThisPrefix
+           type: string
+           stringValue: "kube/"
+         - name: mustBelongToOneOfThese
+           type: stringList
+           stringListValue: [ kube/admins, kube/developers, kube/auditors ]
+         - name: additionalAdmins
+           type: stringList
+           stringListValue: [ ryan@example.com, ben@example.com, josh@example.com ]
+
+       # An optional list of transforms and policies to be executed in the order given during every login attempt.
+       # Each is a CEL expression. It may use the basic CEL language plus the CEL string extensions from cel-go.
+       # The username, groups, and the constants defined above are available as variables in all expressions.
+       # In the first version of this feature, the only allowed types would be policy/v1, username/v1, and groups/v1.
+       # This leaves room for other future possible types and type versions.
+       # Each policy/v1 must return a boolean, and when it returns false, the login is rejected.
+       # Each username/v1 transform must return the new username (a string), which can be the same as the old username.
+       # Each groups/v1 transforms must return the new groups list (list of strings), which can be the same as the old
+       # groups list.
+       # After each expression, the new (potentially changed) username or groups get passed to the following expression.
+       # Any compilation or type-checking failure of any expression will cause an error status on the FederationDomain.
+       # Any unexpected runtime evaluation errors (e.g. division by zero) cause the login to fail.
+       # When all expressions evaluate successfully, then the username and groups has been decided for that login.
+       expressions:
+         # This expression runs first, so it operates on unmodified usernames and groups as extracted from the IDP.
+         # It rejects auth for any user who does not belong to certain groups.
+         - type: policy/v1
+           expression: 'groups.exists(g, g in strListConst.mustBelongToOneOfThese)'
```

Comment:
I think `strListConst` may have drifted from the above items listed in `constants`.

Reply:
I was having a hard time coming up with good names for these CEL variables. Maybe they are not worth abbreviating e.g. `strListConst` instead of `stringListConst` just to save two letters?

Reply:
Let's move this conversation to here: https://github.com/vmware-tanzu/pinniped/pull/1419/files#r1106128924